### ISSUE TITLE
Schema priority between connection and input tools

### ISSUE THREAD

**Issue by simonaubertbd** [amphi-ai/amphi-etl#65]
Hello,

Using the last release with the nice new connection feature.

Here my pipeline and the error messages
![image](https://github.com/user-attachments/assets/c07a1bc9-bc25-47f7-844d-2322468f07ab)

`ERROR: Could not find a version that satisfies the requirement os (from versions: none) ERROR: No matching distribution found for os`

`Error

Execution failed on sql 'SELECT * FROM .articles': ERREUR: erreur de syntaxe sur ou près de « . »
LINE 1: SELECT * FROM .articles
^

[Show Traceback](http://localhost:8888/lab#)

---------------------------------------------------------------------------
SyntaxError Traceback (most recent call last)
File ~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pandas\io\sql.py:2674, in SQLiteDatabase.execute(self, sql, params)
2673 try:
-> 2674 cur.execute(sql, *args)
2675 return cur

SyntaxError: ERREUR: erreur de syntaxe sur ou près de « . »
LINE 1: SELECT * FROM .articles
^

The above exception was the direct cause of the following exception:

DatabaseError Traceback (most recent call last)
Cell In[6], line 26
24 try:
25 with postgresInput1_Engine.connect() as conn:
---> 26 postgresInput1 = pd.read_sql(
27 f"""SELECT * FROM {POSTGRES_SCHEMA}.articles""",
28 con=conn.connection
29 ).convert_dtypes()
30 finally:
31 postgresInput1_Engine.dispose()

File ~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pandas\io\sql.py:706, in read_sql(sql, con, index_col, coerce_float, params, parse_dates, columns, chunksize, dtype_backend, dtype)
704 with pandasSQL_builder(con) as pandas_sql:
705 if isinstance(pandas_sql, SQLiteDatabase):
--> 706 return pandas_sql.read_query(
707 sql,
708 index_col=index_col,
709 params=params,
710 coerce_float=coerce_float,
711 parse_dates=parse_dates,
712 chunksize=chunksize,
713 dtype_backend=dtype_backend,
714 dtype=dtype,
715 )
717 try:
718 _is_table_name = pandas_sql.has_table(sql)

File ~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pandas\io\sql.py:2738, in SQLiteDatabase.read_query(self, sql, index_col, coerce_float, parse_dates, params, chunksize, dtype, dtype_backend)
2727 def read_query(
2728 self,
2729 sql,
(...)
2736 dtype_backend: DtypeBackend | Literal["numpy"] = "numpy",
2737 ) -> DataFrame | Iterator[DataFrame]:
-> 2738 cursor = self.execute(sql, params)
2739 columns = [col_desc[0] for col_desc in cursor.description]
2741 if chunksize is not None:

File ~\AppData\Local\Programs\Python\Python312\Lib\site-packages\pandas\io\sql.py:2686, in SQLiteDatabase.execute(self, sql, params)
2683 raise ex from inner_exc
2685 ex = DatabaseError(f"Execution failed on sql '{sql}': {exc}")
-> 2686 raise ex from exc

DatabaseError: Execution failed on sql 'SELECT * FROM .articles': ERREUR: erreur de syntaxe sur ou près de « . »
LINE 1: SELECT * FROM .articles
^`

Configuration
![image](https://github.com/user-attachments/assets/8d65bfe8-88de-4b2a-8515-5c58f60e3205)
![image](https://github.com/user-attachments/assets/15b12d51-98e6-41ce-be73-ea304924f2d3)

As you can see, I tested without schema in the connection configuration since my idea was to have one connection that I can use with different schemas.

To be clear, if you put the schema in the connection tool, it runs fine
![image](https://github.com/user-attachments/assets/d5b30010-ff58-4afc-815d-b9ecdb9d6035)

And if I put a schema in the connection tool and another one is the postgres input tool, it doesn't care about the one is the postgres tool
![image](https://github.com/user-attachments/assets/be5e7f1e-6cc8-4a22-9371-fbbf46c3bf11)

So what I think : in the connection window, the schema should be named "default schema" and it should work like that
-the schema in the input tool is prioritary but not mandatory (if not defined, connection schema)
-the schema in the connection window is not mandatory and should be renamed default schema (if not defined, database default schema)

to sum it up : schema=coalesce(input_tool_schema, connection_tool_schema)

A good thing with that is that you can use the same connection tool with different schemas for postgresql input tool...  so you can doing like comparing a table in the "dev" schema to one in the "prod" schema or joining a table in the "sales" schema with one in the "crm" schema, in short allowing more use cases without having a connection tool by schema.

Maybe the same applies to output tools, I didn't test it.

Best regards,

Simon

**Comment by tgourdel** [comment]
Okay, just a look at it. 
There is definitely a bug or multiple bugs.
First, the schema for Postgres should not be in the connection, because when looking at the postgres input, we can see the Schema field does not belong to the connection section.
Then the variable is used nonetheless in the code. 
In any case the input should always have priority.
Thanks for reporting and for the suggestion, I'll take a look

**Comment by tgourdel** [comment]
Alright, so I found the bug. It was a silly miss when refactoring the postgres component. The name of the schema field was wrong hence not taking the specified value. It was actually not linked to the connection!
Will be fixed in next version very soon!

**Comment by tgourdel** [comment]
Should be fixed in 0.5.4. Schema field is now taken into account correctly.
